lib/Dialect/FIRRTL/Transforms/InjectDUTHierarchy.cpp

```diff
@@ -28,6 +31,42 @@ struct InjectDUTHierarchy : public InjectDUTHierarchyBase<InjectDUTHierarchy> {
 };
 } // namespace
 
+/// Add an extra level of hierarchy to a hierarchical path that places the
+/// wrapper instance after the DUT.  E.g., this is converting:
+///
+///   firrtl.hierpath [@Top::@dut, @DUT]
+///
+/// Int:
+///
+///   firrtl.hierpath [@Top::@dut, @DUT::@wrapper, @Wrapper]
+static void addHierarchy(HierPathOp path, FModuleOp dut,
+                         InstanceOp wrapperInst) {
+  auto namepath = path.namepath().getValue();
+
+  size_t nlaIdx = 0;
+  while (path.modPart(nlaIdx) != dut.getNameAttr())
+    ++nlaIdx;
+
+  // Add everything up to the DUT.
+  auto front = namepath.take_front(nlaIdx);
+  SmallVector<Attribute> newNamepath(front.begin(), front.end());
+  newNamepath.push_back(
+      hw::InnerRefAttr::get(dut.moduleNameAttr(), wrapperInst.inner_symAttr()));
+
+  // Add the extra level of hierarchy.
+  if (auto dutRef = namepath[nlaIdx].dyn_cast<hw::InnerRefAttr>())
+    newNamepath.push_back(hw::InnerRefAttr::get(
+        wrapperInst.moduleNameAttr().getAttr(), dutRef.getName()));
+  else
+    newNamepath.push_back(
+        FlatSymbolRefAttr::get(wrapperInst.moduleNameAttr().getAttr()));
+
+  // Add anything left over.
+  auto back = namepath.drop_front(nlaIdx + 1);
+  newNamepath.append(back.begin(), back.end());
+  path->setAttr("namepath", ArrayAttr::get(dut.getContext(), newNamepath));
```

Comment:
`path.namepathAttr(ArrayAttr::get(dut.getContext(), newNamepath)))`

Reply:
Will push this in a `!fixup` commit.